### DISCUSSION TITLE
post synch hooks is ignored

### DISCUSSION THREAD

**Discussion by moley** [argoproj/argo-cd#7871] [Q&A]
Hi to all,

I want to trigger  posthook after synch. I have added a hook like the following both into argocd namespace and afterwards now in the namespace were my help application is deployed. The sync was triggred automatically and manually, but the postsynch hook was not triggered nor visible in the synch status result. This is my Job definition: 

`apiVersion: batch/v1
kind: Job
metadata:
  creationTimestamp: null
  name: integration-test
  namespace: argocd
  annotations:
    argocd.argoproj.io/hook: PostSync
    argocd.argoproj.io/hook-delete-policy: HookSucceeded
spec:
  template:
    spec:
      restartPolicy: Never
      containers:
        - name: post-install-job
          image: "registry.access.redhat.com/ubi8/ubi-minimal:latest"              
          command: 
          - /bin/sh
          - -c 
          - |
            echo "WELCOME TO the post installation hook for ArgoCD "
            echo "-------------------------------------------------"
            echo "Here we could start integration tests or health checks"
            echo "..."

            sleep 5
  backoffLimit: 0          
`

So my questions: 
* Which is the correct namespace to add the resource to? 
* Do I have to configure / link the postsync hook in my application.yaml or is it found automatically?

A short hint how to fix this is really welcome. 
Thanks a lot

**Comment by SkPhilipp** [comment]
@moley the way I got it working (note I'm not affiliated with Argo);

> Do I have to configure / link the postsync hook in my application.yaml or is it found automatically?

Sync hooks should be one of your Kubernetes resource definitions that ArgoCD usually picks up, meaning right next to your other definitions like a Kubernetes Deployment, Ingress and so on. Likely in a target repository. If you are using Helm, they would be part of a Helm chart. Alternatively, you can provide them through a plugin.

When you confirm a Sync through the ArgoCD UI, the jobs should be visible next to the other resources momentarily until it completes execution successfully. Moreover you can also find the history of jobs with kubectl.

> Which is the correct namespace to add the resource to?

The namespace can be any of your choice, in my case I used the namespace of the other resources I was deploying.
